client/src/store/useStore.tsx

```diff
@@ -275,4 +275,42 @@ export function useDistributeRoles() {
   });
 }
 
+const filters = ["alive", "dead", "can vote", "all"] as const;
+type Filters = (typeof filters)[number];
+export function usePlayerFilters(): [
+  players: string[],
+  selectedFilter: Filters,
+  allFilters: Record<Filters, string[]>,
+  setFilter: React.Dispatch<
+    React.SetStateAction<"alive" | "dead" | "can vote" | "all">
+  >,
+] {
+  const { game } = useDefiniteGame();
+  const [selectedFilter, setSelectedFilter] = useState<Filters>("all");
+
+  const playerFilters = useMemo(() => {
+    const playerList = game.orderedPlayers.problems
+      ? game.playerList
+      : game?.orderedPlayers.fullList;
+
+    const playerFilters = {
+      alive: playerList.filter((p) => !game.deadPlayers[p]),
+      dead: playerList.filter((p) => game.deadPlayers[p]),
+      "can vote": playerList.filter(
+        (p) => !(game.deadPlayers[p] && game.deadVotes[p]),
+      ),
+      all: playerList,
+    };
+
+    return playerFilters;
+  }, [game.deadPlayers, game.deadVotes, game.playerList, game.orderedPlayers]);
+
+  return [
```

Comment:
as const this.  TS by default assumes all arrays are mutable, so it says "Everything here could be anything else" but as const will make it type it as each item